agent/consul/stream/subscribe.proto

```diff
@@ -2,12 +2,15 @@ syntax = "proto3";
 
 package stream;
 
+import "structs.proto";
+
 service Consul {
     rpc Subscribe(SubscribeRequest) returns (stream Event) {}
 } 
 
 enum Topic {
     ServiceHealth = 0;
+    EndOfSnapshot = 1;
```

Comment:
I really don't think this should be a topic. The semantics of a topic are "a stream of events all about one thing" when we subscribe to `ServiceHealth` topic I think it's odd to use the topic field to mark "end of snapshot" events within the topic.

Imagine if we extend this to allow subscribing to multiple topics in one call, now they each need to send "end of snapshot" messages but we need to know which topic each one is for so we can't use that.

I think we should add an `EndOfSnapshot` message type which is otherwise an empty struct and make it one of the `payload` types below and then type check in the subscriber loop.